### PR TITLE
Allow for multiple clusters to run on the same host

### PR DESCRIPTION
**Pull request by danielin917**
Changes:
   -cluster based unique names
   -new create_network.yml which establishes a private bridge network for each cluster
   -symlink path nesting error fix for mds, nfs, rgw, rbd-mirror
   -consolidated checks into one file in common, we now only check once per host(as opposed to seeing if our   image is running)
   -verbose option in vagrant variables

### CODE REVIEW

File: roles/ceph-common/tasks/docker/checks.yml [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+    ceph_config_keys:
+      - /etc/ceph/{{ cluster }}.client.admin.keyring
+      - /etc/ceph/{{ cluster }}.conf
+      - /etc/ceph/monmap-{{ cluster }}
+      - /etc/ceph/{{ cluster }}.mon.keyring
+      - /etc/ganesha/ganesha.conf
+      - /var/lib/ceph/bootstrap-osd/{{ cluster }}.keyring
+      - /var/lib/ceph/bootstrap-rgw/{{ cluster }}.keyring
+      - /var/lib/ceph/bootstrap-mds/{{ cluster }}.keyring
+
```
Comment (leseb) [inline roles/ceph-common/tasks/docker/checks.yml]: why these 3 empty lines? please just keep one.

File: roles/ceph-osd/tasks/docker/pre_requisite.yml [hunk 2]
```diff
@@ -24,11 +24,13 @@
   tags:
     with_pkg
 
-- name: install epel-release on redhat
+- name: install epel-release on redhat
```
Comment (leseb) [inline roles/ceph-osd/tasks/docker/pre_requisite.yml]: extra space at the end of the line

**Comment by leseb** [comment]
`single_host` must be declared in the appropriate role path `defaults/main.yml`.

File: roles/ceph-common/defaults/main.yml [hunk 3]
```diff
@@ -332,7 +332,7 @@ docker: false
 # by ceph.conf.j2 template. so it should always be defined
 mon_containerized_deployment_with_kv: false
 mon_containerized_deployment: false
-
+single_host: false
```
Comment (leseb) [inline roles/ceph-common/defaults/main.yml]: can we have a bit more explanation on what the option does?

**Comment by leseb** [comment]
Please rebase again, sorry.

File: Vagrantfile [hunk 4]
```diff
@@ -22,6 +22,7 @@ MEMORY         = settings['memory']
 STORAGECTL     = settings['vagrant_storagectl']
 ETH            = settings['eth']
 DOCKER         = settings['docker']
+VERBOSE        = settings['verbose']
```
Comment (leseb) [inline Vagrantfile]: You need to update `vagrant_variables.yml.sample` and `vagrant_variables.yml.atomic` to reflect this config option

**Comment by leseb** [comment]
Just run `generate_group_vars_sample.sh` and we should be good.

**Comment by danielin917** [comment]
group vars script has been run

**Comment by leseb** [comment]
Please rebase

**Comment by danielin917** [comment]
done

**Comment by leseb** [comment]
@danielin917 I know it has been almost 2 weeks but I have this PR https://github.com/ceph/ceph-ansible/pull/815 idle for a while that I'd like to merge. If I merge it, you would have to rebase, and as mentioned in my last comment of this other PR, it might break a couple of things for Docker deployments.

So basically my question is:

* are you fine with me merging https://github.com/ceph/ceph-ansible/pull/815
* then rebase you patch and give another try?

Let me know.

**Comment by danielin917** [comment]
Yea, just comment once it's been merged so I can rebase and do a quick test!

**Comment by leseb** [comment]
@danielin917 great, thanks!

**Comment by danielin917** [comment]
finally rebased and tested, reminder this only works with updated docker images that have cluster base mon maps

**Comment by leseb** [comment]
@danielin917 can you complete your rebase? :)

**Comment by danielin917** [comment]
This rebase got really messy. I think I have it all fixed up like it was before but I don't have a way of checking now that I'm out of the office.

**Comment by leseb** [comment]
That's fine, just check this when you're back at the office and let me know :)

**Comment by dang** [comment]
@leseb Dan's internship is over, so he won't be back in the office.  I'll try and test in a bit.

**Comment by leseb** [comment]
@dang oh I see, thanks for taking over his work ;)

**Comment by alfredodeza** [comment]
I don't understand the reasoning behind supporting something like this. Multiple clusters on the same hosts will surely create all kinds of issues since it is not considered a "normal" scenario for Ceph.

**Comment by leseb** [comment]
@alfredodeza one of the use case is dynamic Ceph provisioning on container platforms. So users that request a Ceph cluster with dedicated resources, they won't take up all system resources so we can collocate multiple cluster to improve hardware utilization.

**Comment by leseb** [comment]
@dang is this one still on your plate?

**Comment by dang** [comment]
@leseb Yes, sorry.  I've had other things come up.  I'll try to get to it later this week.

**Comment by leseb** [comment]
@dang thanks for the update, that's fine, thanks :)

**Comment by leseb** [comment]
@dang any chance to resume this one? Thanks

**Pull request closed without merging**